libs/s25main/controls/ctrlTable.cpp

```diff
@@ -104,6 +104,47 @@ static int Compare(const std::string& a, const std::string& b, ctrlTable::SortTy
                 return h_a - h_b;
             return min_a - min_b;
         }
+        case SRT::Time:
+        {
+            // Sort by time with format h:mm:ss or mm:ss
+            s25util::ClassicImbuedStream<std::istringstream> ss_a(a);
+            s25util::ClassicImbuedStream<std::istringstream> ss_b(b);
+
+            int val_a[3], val_b[3];
+            char c;
+
+            // "h:mm:ss" or "mm:ss"
+            for(int i = 0; i < 3; i++)
```

Comment:
That code is not to pretty and hard to read. How about getting the time in seconds which makes the comparison below trivial: Start with zero, read a number, if succeeded multiply the previous result by 60 and add the number, else ignore
Also note that you can directly do `if(ss >> number)` which IMO reads better  as "if number was read", but that's only my preference

Reply:
Good point, changed it